src/app/repo.service.ts

```diff
@@ -1,22 +1,40 @@
 import { Injectable } from '@angular/core';
-import { Observable, of, map, find } from 'rxjs';
+import { Observable, of, map } from 'rxjs';
 import { Repository } from './repository';
+import { Apollo } from 'apollo-angular';
+import { GET_REPO, GET_VIEWER } from './queries';
 import { REPOSITORIES } from './mock-data';
 
 @Injectable({
   providedIn: 'root'
 })
 export class RepoService {
+  constructor(private apollo: Apollo) {}
+
+  testAuthentication(): Observable<any> {
+    return this.apollo.watchQuery<any>({
+      query: GET_VIEWER,
+      errorPolicy: 'all'
+    }).valueChanges;
+  }
+
   getRepos(): Observable<Repository[]> {
     const repositories = of(REPOSITORIES);
     return repositories;
   }
-
-  getRepo(name: string): Observable<Repository | undefined> {
-    return this.getRepos().pipe(
-      map(repos => 
-        repos.find(repo => repo.name === name))
-    )
+  
+  getRepo(name: any): Observable<Repository> {
```

Comment:
Since the `GET_REPO` expects that `name` is a `string`, would we be able to type the `name` parameter as `string`?

Reply:
Yes! updated thanks